evap/results/exporters.py

```diff
@@ -215,15 +231,18 @@ def write_overall_results(self, evaluations_with_results, course_results_exist):
 
         if course_results_exist:
             # Only query the number of evaluations once and keep track of it here.
-            count_gt_1 = [e.course_evaluations_count > 1 for e in evaluations]
+            count_gt_1 = [e.course_evaluations_count > 1 for e in evaluations]  # type: ignore[attr-defined]
 
             # Borders only if there is a course grade below. Offset by one column
             self.write_empty_row_with_styles(
                 ["default"] + ["border_left_right" if gt1 else "default" for gt1 in count_gt_1]
             )
 
             self.write_cell(_("Evaluation weight"), "bold")
-            weight_percentages = (f"{e.weight_percentage}%" if gt1 else None for e, gt1 in zip(evaluations, count_gt_1))
+            weight_percentages: Generator[str | Any, Any, Any] = (
```

Comment:
Should be `Generator[str | None, None, None]` or `Iterable[str | None]` (see [the docs](https://docs.python.org/3/library/typing.html#typing.Generator))

Reply:
(The comment was resolved but the code was not changed. I unresolved.)